presto-tests/src/test/java/io/prestosql/tests/TestInformationSchemaConnector.java

```diff
@@ -97,6 +97,48 @@ public void testLimit()
         assertQuery("SELECT count(*) FROM (SELECT * FROM test_catalog.information_schema.tables LIMIT 1000)", "VALUES 1000");
     }
 
+    @Test
+    public void testRoleAuthorizationDescriptor()
+    {
+        assertQuery("SELECT count(*) FROM test_catalog.information_schema.role_authorization_descriptors", "VALUES 100");
+        assertQuery("SELECT count(*) FROM test_catalog.information_schema.roles", "VALUES 50");
+        assertQuery("SELECT count(*) FROM test_catalog.information_schema.enabled_roles", "VALUES 50");
+        assertQuery("SELECT count(*) FROM test_catalog.information_schema.applicable_roles", "VALUES 1");
+        assertQuery("SELECT role_name FROM test_catalog.information_schema.role_authorization_descriptors WHERE grantee = 'user5'", "VALUES ('role2')");
+        assertQuery("SELECT grantee FROM test_catalog.information_schema.role_authorization_descriptors WHERE role_name = 'role2'", "VALUES ('user4'), ('user5')");
+
+        assertMetadataCalls(
+                "SELECT count(*) FROM test_catalog.information_schema.role_authorization_descriptors", "VALUES 100",
+                new MetadataCallsCount()
+                        .withRoleGrantCount(1, 0, 0, 0));
+
+        assertMetadataCalls(
+                "SELECT role_name FROM test_catalog.information_schema.role_authorization_descriptors WHERE grantee = 'user5'", "VALUES ('role2')",
+                new MetadataCallsCount()
+                        .withRoleGrantCount(1, 0, 1, 0));
+
+        assertMetadataCalls(
+                "SELECT grantee FROM test_catalog.information_schema.role_authorization_descriptors WHERE role_name = 'role2'", "VALUES ('user4'), ('user5')",
+                new MetadataCallsCount()
+                        .withRoleGrantCount(1, 1, 0, 0));
+
+        assertMetadataCalls(
+                "SELECT grantee FROM test_catalog.information_schema.role_authorization_descriptors WHERE role_name = 'role2' AND grantee = 'user4'", "VALUES 'user4'",
+                new MetadataCallsCount()
+                        .withRoleGrantCount(1, 1, 1, 0));
+
+        assertMetadataCalls(
+                "SELECT count(*) FROM (SELECT * FROM test_catalog.information_schema.role_authorization_descriptors LIMIT 1)", "VALUES 1",
+                new MetadataCallsCount()
+                        .withRoleGrantCount(1, 0, 0, 1));
+
+        // verify that predicate and LIMIT are not pushed down together
+        assertMetadataCalls(
+                "SELECT count(*) FROM (SELECT * FROM test_catalog.information_schema.role_authorization_descriptors WHERE grantee = 'user5' LIMIT 1)", "VALUES 1",
+                new MetadataCallsCount()
+                        .withRoleGrantCount(1, 0, 1, 0));
```

Comment:
That is very nice. How about going one step further and having this more fluent, like 
```
new MetadataCallsCount().with(roleGrantCount(/*method calls count*/ 1).withGranteesPushedDown(1));
```
?

Reply:
I had those methods before. It just seemed to now make more sense to compare the beans...

Maybe it was better the way I had it before?

In the end we have certain metrics we want to count for the connector; those are the atomic counters.
I think I misunderstood what you wanted... MetadataCallsCount is already a "bean", and so having the individual members for counts is fine...?